win32_deps_build.sh

```diff
@@ -0,0 +1,332 @@
+
+function _make() {
+  make -j $NUM_WORKERS $@
+}
+
+mkdir -p $DEPS_DIR
+mkdir -p $depsToolsetDir
+mkdir -p $depsSrcDir
+
+MINGW_CMAKE_FILE="$DEPS_DIR/mingw.cmake"
+cat > $MINGW_CMAKE_FILE <<EOL
+set(CMAKE_SYSTEM_NAME Windows)
+set(TOOLCHAIN_PREFIX x86_64-w64-mingw32)
+
+# We'll need to use posix threads in order to use
+# C++11 features, such as std::thread.
+set(CMAKE_C_COMPILER \${TOOLCHAIN_PREFIX}-gcc-posix)
+set(CMAKE_CXX_COMPILER \${TOOLCHAIN_PREFIX}-g++-posix)
+set(CMAKE_RC_COMPILER \${TOOLCHAIN_PREFIX}-windres)
+
+set(CMAKE_FIND_ROOT_PATH /usr/\${TOOLCHAIN_PREFIX} /usr/lib/gcc/\${TOOLCHAIN_PREFIX}/7.3-posix)
+set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
+set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY BOTH)
+set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE BOTH)
+EOL
+
+sudo apt-get -y install mingw-w64 cmake pkg-config python3-dev python3-pip \
+                autoconf libtool ninja-build
+sudo python3 -m pip install cython
+
+cd $depsSrcDir
+if [[ ! -d $zlibDir ]]; then
+    git clone https://github.com/madler/zlib
+fi
+cd zlib
+# Apparently the configure script is broken...
+sed -e s/"PREFIX ="/"PREFIX = x86_64-w64-mingw32-"/ -i win32/Makefile.gcc
+_make -f win32/Makefile.gcc
+_make BINARY_PATH=$zlibDir \
+     INCLUDE_PATH=$zlibDir/include \
+     LIBRARY_PATH=$zlibDir/lib \
+     SHARED_MODE=1 \
+     -f win32/Makefile.gcc install
+
+cd $depsToolsetDir
+if [[ ! -d $lz4Dir ]]; then
+    git clone https://github.com/lz4/lz4
+fi
+cd lz4
+git checkout $lz4Tag
+_make BUILD_STATIC=no CC=x86_64-w64-mingw32-gcc \
+      DLLTOOL=x86_64-w64-mingw32-dlltool OS=Windows_NT
+
+cd $depsSrcDir
+if [[ ! -d $sslSrcDir ]]; then
+    curl "https://www.openssl.org/source/openssl-${sslVersion}.tar.gz" | tar xz
+fi
+cd $sslSrcDir
+mkdir -p $sslDir
+CROSS_COMPILE="x86_64-w64-mingw32-" ./Configure \
+    mingw64 shared --prefix=$sslDir
+_make depend
+_make
+_make install
+
+cd $depsSrcDir
+if [[ ! -d $curlSrcDir ]]; then
+    git clone https://github.com/curl/curl
+fi
+cd $curlSrcDir
+git checkout $curlTag
+./buildconf
+./configure --prefix=$curlDir --with-ssl=$sslDir --with-zlib=$zlibDir \
+            --host=x86_64-w64-mingw32
+_make
+_make install
+
+
+cd $depsSrcDir
+if [[ ! -d $boostSrcDir ]]; then
+    wget -qO- $boostUrl | tar xz
+fi
+
+cd $boostSrcDir
+echo "using gcc : mingw32 : x86_64-w64-mingw32-g++-posix ;" > user-config.jam
+
+# Workaround for https://github.com/boostorg/thread/issues/156
+# Older versions of mingw provided a different pthread lib.
+sed -i 's/lib$(libname)GC2.a/lib$(libname).a/g' ./libs/thread/build/Jamfile.v2
+sed -i 's/mthreads/pthreads/g' ./tools/build/src/tools/gcc.py
+sed -i 's/mthreads/pthreads/g' ./tools/build/src/tools/gcc.jam
+
+sed -i 's/pthreads/mthreads/g' ./tools/build/src/tools/gcc.py
+sed -i 's/pthreads/mthreads/g' ./tools/build/src/tools/gcc.jam
+
+export PTW32_INCLUDE=/usr/share/mingw-w64/include
+export PTW32_LIB=/usr/x86_64-w64-mingw32/lib
+
+# Fix getting Windows page size
+cat > thread_data.patch <<EOL
```

Comment:
You could pipe it directly into patch, instead of making a intermediate file.
`cat | patch ......... << EOL`

Reply:
Makes sense. To be honest, having those patch calls one next to the other made it a bit easier to see which specific files are being patched. Probably not worth using those intermediate files just for that.